packages/devextreme/testing/tests/DevExpress.ui.widgets/chat.tests.js

```diff
@@ -134,6 +146,44 @@ QUnit.module('Chat initialization', moduleConfig, () => {
 
         assert.strictEqual(invalidateStub.callCount, 1);
     });
+
+    QUnit.test('Last message should be into view', function(assert) {
+        const messages = generateMessages(31);
+
+        this.instance.option({ items: messages });
+
+        const $messages = this.$element.find(`.${CHAT_MESSAGE_BUBBLE_CLASS}`);
+        const lastMessage = $messages[$messages.length - 1];
+        const lastMessageOffset = Math.ceil($(lastMessage).offset().top);
+
+        assert.strictEqual(lastMessageOffset, -9459);
```

Comment:
It is completely unclear what we are testing with this test and why we are using these magic numbers. According to the test name 'Last message should be into view,' we are checking that the element will be visible, but it cannot be visible because the element is absolutely positioned and located off-screen.

It would make more sense to check that, after initialization or after a message is entered, we have scrolled to the maximum position.

Reply:
[5f471b926b7031dac4c622c538f731e7447e4126]

In the first new two tests, we can't override the scrollable scrollToElement method on initialization (just like I did in the last test). So the only way to test the performance is to get the value of scrollTop()